firebase-config/src/main/java/com/google/firebase/remoteconfig/internal/rollouts/RolloutsStateSubscriptionsHandler.java

```diff
@@ -16,19 +16,73 @@
 
 package com.google.firebase.remoteconfig.internal.rollouts;
 
+import static com.google.firebase.remoteconfig.FirebaseRemoteConfig.TAG;
+
+import android.util.Log;
 import androidx.annotation.NonNull;
+import com.google.firebase.remoteconfig.FirebaseRemoteConfigException;
+import com.google.firebase.remoteconfig.internal.ConfigCacheClient;
 import com.google.firebase.remoteconfig.internal.ConfigContainer;
+import com.google.firebase.remoteconfig.interop.rollouts.RolloutsState;
 import com.google.firebase.remoteconfig.interop.rollouts.RolloutsStateSubscriber;
+import java.util.Collections;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Executor;
 
 public class RolloutsStateSubscriptionsHandler {
+  private ConfigCacheClient activatedConfigsCache;
+  private RolloutsStateFactory rolloutsStateFactory;
+  private Executor executor;
+
+  public RolloutsStateSubscriptionsHandler(
+      @NonNull ConfigCacheClient activatedConfigsCache,
+      @NonNull RolloutsStateFactory rolloutsStateFactory,
+      @NonNull Executor executor) {
+    this.activatedConfigsCache = activatedConfigsCache;
+    this.rolloutsStateFactory = rolloutsStateFactory;
+    this.executor = executor;
+  }
 
-  public RolloutsStateSubscriptionsHandler() {}
+  // Thread-safe implementation for subscribers, using a ConcurrentHashMap as the underlying
+  // implementation with set-like accessors.
+  private Set<RolloutsStateSubscriber> subscribers =
```

Comment:
Could we initialize this in the constructor?

Reply:
We could - I think it's more readable to declare and initialize the field at the same time if it's a simple initialization, though, so would prefer to keep as-is. I moved it up above the constructor though

Reply:
Cool